### PR TITLE
adapt FilterPartLabel

### PR DESCRIPTION
**Pull request by jensdrenhaus**
Complement the "Filter...()" set of methods with `func (b BlockDevices) FilterPartLabel(label string) BlockDevices` using information from a `gpt.Table` according to the other methods instead of reading udev's `/dev/disk/by-partlabel/` list. The latter is usually not available in u-root.

Signed-off-by: Jens Drenhaus <jens.drenhaus@9elements.com>

### CODE REVIEW

File: pkg/mount/block/blockdev.go [hunk 1]
```diff
@@ -498,6 +498,32 @@ func (b BlockDevices) FilterPartType(guid string) BlockDevices {
...
+	for _, device := range b {
+		table, err := device.GPTTable()
+		if err != nil {
+			continue
+		}
+		for i, part := range table.Partitions {
+			if part.IsEmpty() {
+				continue
+			}
+			if strings.ToLower(part.Name()) == strings.ToLower(label) {
```
Comment (rjoleary) [inline pkg/mount/block/blockdev.go:514]: Can you use this function which is a bit more robust? https://golang.org/pkg/strings/#EqualFold

File: pkg/mount/block/blockdev.go [hunk 2]
```diff
@@ -498,6 +498,32 @@ func (b BlockDevices) FilterPartType(guid string) BlockDevices {
 	return b.FilterNames(names...)
 }
 
+// FilterPartLabel returns a list of BlockDev objects whose underlying block
+// device has the given parition label.
```
Comment (rjoleary) [inline pkg/mount/block/blockdev.go:502]: Add a comment saying this is case-insensitive.

File: pkg/mount/block/blockdev.go [hunk 3]
```diff
@@ -498,6 +498,32 @@ func (b BlockDevices) FilterPartType(guid string) BlockDevices {
...
+		table, err := device.GPTTable()
+		if err != nil {
+			continue
+		}
+		for i, part := range table.Partitions {
+			if part.IsEmpty() {
+				continue
+			}
+			if strings.ToLower(part.Name()) == strings.ToLower(label) {
+				r := []rune(device.Name[len(device.Name)-1:])
```
Comment (rjoleary) [inline pkg/mount/block/blockdev.go:515]: Won't this break once you go beyond 10 partitions?

File: pkg/mount/block/blockdev.go [hunk 4]
```diff
@@ -498,6 +498,32 @@ func (b BlockDevices) FilterPartType(guid string) BlockDevices {
...
+				if unicode.IsDigit(r[0]) {
+					names = append(names, fmt.Sprintf("%sp%d", device.Name, i+1))
```
Comment (rjoleary) [inline pkg/mount/block/blockdev.go:517]: I'm a bit confused by what is going on here. Are you assuming a device ending in a number (ex: nvme0n1) have partitions numbered like nvme0n1p1, nvme0n1p2, ... and devices ending in a letter (ex: sda) have partitions numbered like sda1, sda2, ...? Perhaps add a comment.

**Review by rjoleary — COMMENTED** [review]
Can you add a description to the commit message?

File: pkg/mount/block/blockdev.go (hunk 1 above)
Comment (jensdrenhaus) [reply pkg/mount/block/blockdev.go:514]: Sure. I took this from the already existing similar functions. Shall I change it in FilterPartID and FilterPartType, too?

File: pkg/mount/block/blockdev.go (hunk 4 above)
Comment (jensdrenhaus) [reply pkg/mount/block/blockdev.go:517]: Yes, I guess so. Took that part from the existing functions FilterPartID and FilterPartType.

File: pkg/mount/block/blockdev.go (hunk 3 above)
Comment (jensdrenhaus) [reply pkg/mount/block/blockdev.go:515]: Oh, that's true. Should this case be covered. Or should we add a comment. Same goes for FilterPartID and FilterPartType.

File: pkg/mount/block/blockdev.go (hunk 1 above)
Comment (rjoleary) [reply pkg/mount/block/blockdev.go:514]: @hugelgupf I see you authored this line originally. Was there any good reason not to use EqualFold?

File: pkg/mount/block/blockdev.go (hunk 3 above)
Comment (rjoleary) [reply pkg/mount/block/blockdev.go:515]: At minimum skip the partition if the number is >10. Right now, it looks like it'll try to boot from the wrong partition in this case.

File: pkg/mount/block/blockdev.go (hunk 4 above)
Comment (rjoleary) [reply pkg/mount/block/blockdev.go:517]: @rminnich Do you know if this is always consistent with how Linux numbers partitions?

Comment (jensdrenhaus) [reply pkg/mount/block/blockdev.go:526]: ping @rminnich

File: pkg/mount/block/blockdev.go (hunk 1 above)
Comment (jensdrenhaus) [reply pkg/mount/block/blockdev.go:523]: ping @hugelgupf

File: pkg/mount/block/blockdev.go (hunk 3 above)
Comment (jensdrenhaus) [reply pkg/mount/block/blockdev.go]: I've moved that part into a helper function and added a test. Seems to work out with higher partition numbers. If I miss anything here, let me know.

File: pkg/mount/block/blockdev.go [hunk 5]
```diff
@@ -485,13 +493,29 @@ func (b BlockDevices) FilterPartType(guid string) BlockDevices {
...
+			if strings.EqualFold(part.Type.String(), guid) {
+				names = append(names, composePartName(device.Name, i+1))
+			}
+		}
+	}
+	return b.FilterNames(names...)
+}
+
+// FilterPartLabel returns a list of BlockDev objects whose underlying block
+// device has the given parition label. The name comparison is case-insensitive.
```
Comment (rjoleary) [inline pkg/mount/block/blockdev.go:505]:
```suggestion
// device has the given partition label. The name comparison is case-insensitive.
```

File: pkg/mount/block/blockdev_test.go [hunk 6]
```diff
@@ -110,3 +110,81 @@ func TestParsePCIBlockList(t *testing.T) {
 		})
 	}
 }
+
+func TestComposePartName(t *testing.T) {
+
```
Comment (rjoleary) [inline pkg/mount/block/blockdev_test.go:115]: Remove empty line.

**Review by rjoleary — APPROVED** [review]
Two small nits; otherwise, this looks good.

**Comment by jensdrenhaus** [comment]
Sorry for force-pushing, need your approval again @rjoleary. The two small nits you mentioned are included.

**Review by rjoleary — APPROVED** [review]

**Comment by rjoleary** [comment]
> Sorry for force-pushing, need your approval again @rjoleary. The two small nits you mentioned are included.

Thank you! Sorry for the delay

**Pull request merged**